name: bug_tracker
services:
  db:
    image: postgres:17
    restart: always
    environment:
      - POSTGRES_DB=${DATABASE_NAME}
      - POSTGRES_USER=${DATABASE_USER}
      - POSTGRES_PASSWORD=${DATABASE_PASSWORD}
      - POSTGRES_SERVER=${DATABASE_SERVER}
      - PGDATA=/var/lib/postgresql/data
    ports:
      - "5432:5432"
    networks:
      - app_network
  web:
    build:
      context: .
      dockerfile: ./docker/django/Dockerfile
    working_dir: /app/bug_tracker
    volumes:
      - .:/app
    expose:
      - 8000
    depends_on:
      - db
    networks:
    - app_network
  nginx:
    image: nginx:1.27
    volumes:
      - ./media:/app/media
      - ./staticfiles:/app/staticfiles
    depends_on:
      - web
    networks:
      - app_network
  redis:
    image: redis:7-alpine
    restart: always
    ports:
      - "6379:6379"
    networks:
      - app_network
  celery:
    build:
      context: .
      dockerfile: ./docker/django/Dockerfile
    command: celery -A bug_tracker worker --loglevel INFO
    working_dir: /app/bug_tracker
    depends_on:
      - redis
      - web
    volumes:
      - .:/app
    networks:
      - app_network
  flower:
    build:
      context: .
      dockerfile: ./docker/django/Dockerfile
    working_dir: /app/bug_tracker
    command: celery -A bug_tracker flower --port=5555 --loglevel INFO
    depends_on:
      - redis
      - celery
    ports:
      - "5555:5555"
    networks:
      - app_network
networks:
  app_network:
    driver: bridge
